browser/brave_wallet/asset_discovery_task_unittest.cc

```diff
@@ -1614,7 +1615,8 @@ TEST_F(AssetDiscoveryTaskUnitTest, DiscoverNFTs) {
   GURL url;
 
   // Empty addresses yields empty expected_contract_addresses
-  wallet_service_->SetNftDiscoveryEnabled(true);
+  GetPrefs()->SetBoolean(kBraveWalletNftDiscoveryEnabled, true);
```

Comment:
with `PrefChangeRegistrar` you don't have to do these changes which looks suspicious